### PR TITLE
Mesh 965/submodule settlement

### PR DESCRIPTION
**Pull request by maxsam4**
Please also review:
https://github.com/PolymathNetwork/polymesh-settlement-module/pull/1

### CODE REVIEW

File: pallets/runtime/common/tests/all/settlement_test.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+../../../../settlement/src/tests.rs
```
Comment (vkomenda) [inline pallets/runtime/common/tests/all/settlement_test.rs:1]: What is the purpose of this file? Is it supposed to be a symbolic link?

Comment (maxsam4) [reply pallets/runtime/common/tests/all/settlement_test.rs:1]: Yes, this is a symbolic link to the test file in the settlement submodule.

Comment (maxsam4) [reply pallets/runtime/common/tests/all/settlement_test.rs:1]: Funny that GitHub shows it this way :)

File: pallets/asset/src/lib.rs [hunk 2]
```diff
@@ -2244,4 +2305,81 @@ impl<T: Trait> Module<T> {
...
+        let updated_to_balance = receiver_balance.saturating_add(value);
+
+        Self::_update_checkpoint(ticker, from_did, sender_balance);
+        Self::_update_checkpoint(ticker, to_did, receiver_balance);
+
+        // reduce sender's balance
+        <BalanceOf<T>>::insert(ticker, &from_did, updated_from_balance);
+
+        // increase receiver's balance
+        <BalanceOf<T>>::insert(ticker, &to_did, updated_to_balance);
```
Comment (satyamakgec) [inline pallets/asset/src/lib.rs:2373]: Here, if `to_did == from_did` then it will affect the calculation. The same bug pointed out by the Pablo.

Comment (maxsam4) [reply pallets/asset/src/lib.rs:2373]: I'll add a check in the settlement module to make sure this never happens. I don't want this function to revert in any case so I'll let this be like this.

**Pull request merged**